shinysdr/plugins/rtty/__init__.py

```diff
@@ -55,59 +56,105 @@
+    '''
+
     implements(IDemodulator)
     
-    __filter_low = 1500
-    __filter_high = 2500
-    __transition = 100
+    _spacing = 170
+
+    _low_cutoff = _spacing * -2
+    _high_cutoff = _spacing
+    _transition_width = _spacing
 
-    def __init__(self, mode,
-            input_rate=0,
-            context=None):
+    def __init__(self, mode, input_rate=0, context=None):
         assert input_rate > 0
+        self.input_rate = input_rate
         gr.hier_block2.__init__(
             self, 'RTTY demodulator',
             gr.io_signature(1, 1, gr.sizeof_gr_complex * 1),
             gr.io_signature(1, 1, gr.sizeof_float * 1),
         )
-        self.__text = u''
-        
-        baud = _DEFAULT_BAUD  # TODO param
-        self.baud = baud
 
-        demod_rate = 6000  # TODO optimize this value
-        self.samp_rate = demod_rate  # TODO rename
+        self.samp_rate = 6000
         
-        self.__channel_filter = MultistageChannelFilter(
-            input_rate=input_rate,
-            output_rate=demod_rate,
-            cutoff_freq=self.__filter_high,
-            transition_width=self.__transition)  # TODO optimize filter band
-        self.__sharp_filter = grfilter.fir_filter_ccc(
-            1,
-            firdes.complex_band_pass(1.0, demod_rate,
-                self.__filter_low,
-                self.__filter_high,
-                self.__transition,
-                firdes.WIN_HAMMING))
-        self.fsk_demod = RTTYFSKDemodulator(input_rate=demod_rate, baud=baud)
-        self.__real = blocks.complex_to_real(vlen=1)
+        channel_filter = self._make_channel_filter()
+
+        self.__text = u''
         self.__char_queue = gr.msg_queue(limit=100)
         self.char_sink = blocks.message_sink(gr.sizeof_char, self.__char_queue, True)
 
         self.connect(
             self,
-            self.__channel_filter,
-            self.__sharp_filter,
-            self.fsk_demod,
-            rtty.rtty_decode_ff(rate=demod_rate, baud=baud, polarity=False),
+            channel_filter,
+            self._make_demodulator(),
             self.char_sink)
         
         self.connect(
-            self.__sharp_filter,
-            self.__real,
-            self)
-    
+            channel_filter,
+            self._make_audio_filter(),
+            blocks.rotator_cc(rotator_inc(self.samp_rate, 2000 + self._spacing / 2)),
+            blocks.complex_to_real(vlen=1),
+            analog.agc2_ff(
+                reference=dB(-10),
+                attack_rate=8e-1,
+                decay_rate=8e-1,
+            ),
+            self,
+        )
+
+    def _make_demodulator(self):
+        return rtty_demod_cb(
+            samp_rate=self.samp_rate,
+            # 6000 / 45.45 / 11 = 12.0012 samples per bit. We want a number
+            # that's not too small to avoid quantization errors in the
+            # timing.
+            decimation=11,
+            mark_freq=0,
+            space_freq=-self._spacing,
+        )
```

Comment:
move `)` to previous line here and below

(Yes, the __init__ call is not consistent with this; I used to have a less clear idea of code style and stuff like that is left around. Change it or leave it as you feel like.)